mypy/build.py

```diff
@@ -808,20 +862,38 @@ def find() -> Optional[str]:
         # that will require the same subdirectory.
         components = id.split('.')
         dir_chain = os.sep.join(components[:-1])  # e.g., 'foo/bar'
-        if (dir_chain, lib_path) not in find_module_dir_cache:
-            dirs = []
-            for pathitem in lib_path:
-                # e.g., '/usr/lib/python3.4/foo/bar'
-                isdir = find_module_isdir_cache.get((pathitem, dir_chain))
-                if isdir is None:
-                    dir = os.path.normpath(os.path.join(pathitem, dir_chain))
-                    isdir = os.path.isdir(dir)
-                    find_module_isdir_cache[pathitem, dir_chain] = isdir
-                if isdir:
-                    dirs.append(dir)
-            find_module_dir_cache[dir_chain, lib_path] = dirs
-        candidate_base_dirs = find_module_dir_cache[dir_chain, lib_path]
-
+        if dir_chain:
+            if dir_chain not in find_module_dir_cache:
+                dirs = []
+
+                # Regular packages on the PATH
+                for pathitem in lib_path:
+                    # e.g., '/usr/lib/python3.4/foo/bar'
+                    isdir = find_module_isdir_cache.get((pathitem, dir_chain))
+                    if isdir is None:
+                        dir = os.path.normpath(os.path.join(pathitem, dir_chain))
+                        isdir = os.path.isdir(dir)
+                        find_module_isdir_cache[pathitem, dir_chain] = isdir
+                    if isdir:
+                        dirs.append(dir)
+
+                # Third-party stub/typed packages
+                for pkg_dir in package_dirs:
+                    stub_name = components[0] + '_stubs'
+                    typed_file = os.path.join(pkg_dir, components[0], 'py.typed')
+                    stub_typed_file = os.path.join(pkg_dir, stub_name, 'py.typed')
```

Comment:
It wasn't clear to me from reading PEP 561 that stub packages also need a py.typed file.

Reply:
Hm, I intended the section saying
>Package maintainers who wish to support type checking of their code MUST add a marker file named py.typed to their package supporting typing.
to mean *all* packages should add it (in my mind including stub only packages). But I suppose having the `_stubs` suffix is rather indicative that it supports typing. I don't entirely have a preference here, but it seems more consistent to have the file in stub packages.